### PR TITLE
Integration tests for Logs

### PR DESCRIPTION
**Pull request by lalitb**
## Changes

This is ready for review. It tests the basic use-case of OTLP Log export using default (grpc) exporter, and compare the generated log file, with one stored in the repo.
The changes are inspired (actually shameless copy :) ) from @TommyCpp 's implementation for span's integration tests.

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [x] Unit tests added/updated (if applicable)
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [x] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: opentelemetry-otlp/tests/integration_test/Cargo.toml [hunk 1]
```diff
@@ -12,7 +12,8 @@ opentelemetry_sdk = { path = "../../../opentelemetry-sdk", features = ["rt-tokio
 opentelemetry-otlp = { path = "../../../opentelemetry-otlp", features = ["tonic", "metrics", "logs"] }
 opentelemetry-semantic-conventions = { path = "../../../opentelemetry-semantic-conventions" }
 opentelemetry-proto = { path = "../../../opentelemetry-proto", features = ["gen-tonic-messages", "trace", "with-serde"] }
+opentelemetry-appender-log = { path = "../../../opentelemetry-appender-log", default-features = false}
+log = { workspace = true }
 tokio = { version = "1.0", features = ["full"] }
 serde_json = "1"
 testcontainers = "0.15.0"
-# env_logger = "0.11.3" // uncomment if needed for local debugging.
```
Comment (lalitb) [inline opentelemetry-otlp/tests/integration_test/Cargo.toml:18]: This won't be usable further for local debugging. As we can only have one subscriber with `log`.

File: opentelemetry-proto/tests/grpc_build.rs [hunk 2]
```diff
@@ -94,8 +94,10 @@ fn build_tonic() {
     }
 
     // add custom serializer and deserializer for AnyValue
-    builder = builder
-        .field_attribute("common.v1.KeyValue.value", "#[cfg_attr(feature =\"with-serde\", serde(serialize_with = \"crate::proto::serializers::serialize_to_value\", deserialize_with = \"crate::proto::serializers::deserialize_from_value\"))]");
+    for path in ["common.v1.KeyValue.value", "logs.v1.LogRecord.body"] {
+        builder = builder
+        .field_attribute(path, "#[cfg_attr(feature =\"with-serde\", serde(serialize_with = \"crate::proto::serializers::serialize_to_value\", deserialize_with = \"crate::proto::serializers::deserialize_from_value\"))]");
+    }
```
Comment (lalitb) [inline opentelemetry-proto/tests/grpc_build.rs:100]: LogRecord::body is `AnyValue`, and we need a custom serializer/deserializer for it.

Comment (TommyCpp) [reply opentelemetry-proto/tests/grpc_build.rs:100]: Yep that's a good follow up but not necessary blocking

**Review by TommyCpp — APPROVED** [review]

File: opentelemetry-proto/tests/grpc_build.rs (hunk 2 above)
Comment (lalitb) [reply opentelemetry-proto/tests/grpc_build.rs:100]: Thanks, this has been added to this PR, as was a pre-requisite to serialize/deserialize LogData.

**Pull request merged**